plugins/processors/template/template.go

```diff
@@ -0,0 +1,38 @@
+package template
+
+import (
+	"github.com/influxdata/telegraf"
+	"strings"
+	"text/template"
+)
+
+type TemplateProcessor struct {
+	Tag      string `toml:"tag"`
+	Template string `toml:"template"`
+}
+
+func (r *TemplateProcessor) Apply(in ...telegraf.Metric) []telegraf.Metric {
+	// create template
+	tmpl, err := template.New("test").Parse(r.Template)
+	if err != nil {
+		panic(err)
+	}
+
+	// for each metric in "in" array
+	for _, metric := range in {
+		var b strings.Builder
+		//newM := TemplateMetric{metric}
+
+		// supply TemplateMetric and Template from configuration to Template.Execute
+		err := tmpl.Execute(&b, metric)
+		if err != nil {
+			panic(err)
+		}
+
+		metric.AddTag(r.Tag, b.String())
+	}
+
+	// convert/wrap metric in TemplateMetric
+	// convert TemplateMetric back to metric?
```

Comment:
You won't need to do this, `return in` will be fine.

Reply:
I tried removing the AddTag call and the unit test now fails. Presumably the template Execute has no way of adding a Tag.
Or is that something I should add to TemplateMetric?